### PR TITLE
added get articles with comment count

### PR DESCRIPTION
**Pull request by Gary-Huynh**

### CODE REVIEW

File: __tests__/app.test.js [hunk 1]
```diff
@@ -87,4 +87,27 @@ describe("404: Not Found",()=>{
...
     })
+})
+
+
+describe("GET /api/articles",()=>{
+
+    test("200: /api/articles should return an object with all current articles",()=>{
+        return request(app).get('/api/articles').expect(200)
+        .then(({body})=>{
+            expect(body.articles.length).toBeGreaterThan(0)
```
Comment (Dutton1990) [inline __tests__/app.test.js]: Remember that we are dealing with test data and we are reseeding the db after each test - you can be more specific here. How many articles are there?

File: model.js [hunk 2]
```diff
@@ -24,6 +23,12 @@ exports.selectSpecificArticle = (article_id)=>{
     }
     return article.rows
  })
+}
 
+exports.selectAllArticles = ()=>{
 
+    return db.query('SELECT articles.author, title, articles.article_id, topic, articles.created_at,articles.votes,article_img_url, COUNT(comments.body) AS comment_count FROM articles JOIN comments ON articles.article_id = comments.article_id GROUP BY articles.article_id ORDER BY created_at DESC;')
+    .then((articles)=>{
```
Comment (Dutton1990) [inline model.js:36]: Nice job with the query!

**Comment by Dutton1990** [comment]
Good start Gary, but we have a few more things to add 😄 

The task wants some additional features - 
In addition:

the articles should be sorted by date in descending order.
Consider what errors could occur with this endpoint, and make sure to test for them.

You will need more tests to prove how they are sorted and what happens if you try to sort by values that don't match. 

When you've added the above, please resubmit this PR on nchelp pr and one of us will check it 👍

**Review by RichardNeat — APPROVED** [review]
Nice one Gary :) merge away!

**Pull request merged**